src/odemis/gui/cont/acquisition.py

```diff
@@ -1027,6 +1028,171 @@ def open_acquisition_dialog(self):
+
+        # TODO: the button should be disabled as long as no feature is selected (with a message "select a feature")
+        feature = self._tab_data.main.currentFeature.value
+        if feature is None:
+            raise ValueError("Select a feature first to specify the Z localization in X/Y")
+        pos = feature.pos.value[:2]
+
+        # Disable the GUI and show the progress bar
+        self._tab.streambar_controller.pauseStreams()
+        self._tab.streambar_controller.pause()
+
+        self._panel.lbl_z_localization.Hide()
+        self._panel.gauge_z_localization.Show()
+        self._tab_data.main.is_acquiring.value = True
+        self._panel.Layout()
+
+        # Store the acquisition somewhere, for debugging purposes
+        acq_conf = conf.get_acqui_conf()
+        fn = create_filename(acq_conf.pj_last_path, "{datelng}-{timelng}-superz", ".ome.tiff")
+        assert fn.endswith(".ome.tiff")
+
+        # The angles of stigmatorAngle should come from MD_CALIB, so it's relatively safe
+        angle = self._tab_data.stigmatorAngle.value
+
+        # TODO: Convert the "locate Z" button to "cancel", once the function supports cancelling
+        self._acq_future = z_localization.measure_z(self._stigmator, angle, pos, s, logpath=fn)
+        self._acq_future_connector = ProgressiveFutureConnector(self._acq_future,
+                                                                self._panel.gauge_z_localization)
+        self._acq_future.add_done_callback(self._on_measure_z_done)
+
+    @call_in_wx_main
+    def _on_measure_z_done(self, f):
+        """
+        Called when measure_z() is completed (can also happen if cancelled or failed)
+        """
+        try:
+            zshift, warning = f.result()
+
+            # focus position: the base for the shift computed by the z localization
+            zpos_acq = self._tab_data.main.focus.position.value["z"]
+
+            logging.debug("Feature located at %s + %s m", zpos_acq, zshift)
+            zpos = zpos_acq + zshift
+
+            # Sanity check: typically, the Z localization is for localization within a few µm.
+            if abs(zshift) > 100e-6:
+                warning = 7
+
+            # Update the feature Z pos, and move there
+            feature = self._tab_data.main.currentFeature.value
+            pos = feature.pos.value[:2]
+            feature.pos.value = pos + (zpos,)
+            if warning:
+                # Update the Z pos, but do not move there.
+                logging.warning("Z pos shift detected of %s, but not going there as it had warning %s", zshift, warning)
+                popup.show_message(self._tab.main_frame, "Z localization unreliable",
+                                   "The Z localization could not locate the depth with sufficient certainty.",
+                                   level=logging.WARNING)
+            else:
+                f = self._tab_data.main.focus.moveAbs({"z": zpos})
```

Comment:
Don't we want any user interference here? Maybe a small check or something?
Maybe this is being to careful but for example for Meteor this move isn't needed.

Reply:
Before that, we do check that the distance is not too big (100µm), for safety. I'm not sure what's exactly needed for the METEOR. I think there is some merit to still move to the focus position in the METEOR, so that the user can easily confirm that the value is correct (eg, playing the stream and seeing the feature blurry would show the procedure went wrong). For the METEOR, the whole workflow is not yet fully tested anyway, so let's revisit this whole code once it's tested and everyone agrees on a good behaviour.